Assets/Tests/Runtime/NetworkedMessageChannelTests.cs

```diff
@@ -0,0 +1,250 @@
+using System;
+using System.Collections;
+using NUnit.Framework;
+using Unity.Multiplayer.Samples.BossRoom.Shared.Infrastructure;
+using Unity.Netcode;
+using UnityEngine;
+using Unity.Netcode.TestHelpers.Runtime;
+using UnityEngine.TestTools;
+using Assert = UnityEngine.Assertions.Assert;
+
+namespace Unity.Multiplayer.Samples.BossRoom.Tests.Runtime
+{
+    public class NetworkedMessageChannelTests : NetcodeIntegrationTest
+    {
+        struct EmptyMessage : INetworkSerializeByMemcpy { }
+
+        struct GenericMessage : INetworkSerializeByMemcpy
+        {
+            public bool value;
+        }
+
+        protected override int NumberOfClients => 2;
+
+        DisposableGroup m_Subscriptions;
+
+        DIScope[] m_ClientScopes;
+        DIScope m_ServerScope;
+
+        int m_NbMessagesReceived;
+
+        protected override IEnumerator OnSetup()
+        {
+            m_NbMessagesReceived = 0;
+            return base.OnSetup();
+        }
+
+        protected override void OnServerAndClientsCreated()
+        {
+            m_ClientScopes = new DIScope[NumberOfClients];
+            for (int i = 0; i < NumberOfClients; i++)
+            {
+                m_ClientScopes[i] = new DIScope();
+                m_ClientScopes[i].BindInstanceAsSingle(m_ClientNetworkManagers[i]);
+                m_ClientScopes[i].FinalizeScopeConstruction();
+            }
+
+            m_ServerScope = new DIScope();
+            m_ServerScope.BindInstanceAsSingle(m_ServerNetworkManager);
+            m_ServerScope.FinalizeScopeConstruction();
+
+            base.OnServerAndClientsCreated();
+        }
+
+        protected override IEnumerator OnTearDown()
+        {
+            m_Subscriptions.Dispose();
+            for (int i = 0; i < NumberOfClients; i++)
+            {
+                m_ClientScopes[i].Dispose();
+            }
+            m_ServerScope.Dispose();
+            return base.OnTearDown();
+        }
+
+        void InitializeNetworkedMessageChannels<T>(int nbClients, int nbSubscribers, T expectedValue, out NetworkedMessageChannel<T>[] emptyMessageChannelClients, out NetworkedMessageChannel<T> emptyMessageChannelServer) where T : unmanaged, INetworkSerializeByMemcpy
+        {
+            emptyMessageChannelClients = new NetworkedMessageChannel<T>[nbClients];
```

Comment:
wait, how do you associate your message channel with your client and server NetworkManager? NMC uses NetworkManager's Singleton...

Reply:
nvm, you inject it